.buildkite/scripts/publish.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -euo pipefail
+
+source .buildkite/scripts/tooling.sh
+
+pushDockerImage() {
+    docker build \
+        -t "${DOCKER_IMG_TAG}" \
+        --label BRANCH_NAME="${GITHUB_PR_BRANCH}" \
+        --label GIT_SHA="${BUILDKITE_COMMIT}" \
+        --label GO_VERSION="${SETUP_GOLANG_VERSION}" \
+        --label TIMESTAMP="$(date +%Y-%m-%d_%H:%M)" \
+        .
+    retry 3 docker push "${DOCKER_IMG_TAG}"
+    retry 3 docker tag "${DOCKER_IMG_TAG}" "${DOCKER_IMG_TAG_BRANCH}"
```

Comment:
nit. This command is not needed to run retry. 
If it fails once, it will always fail.

```suggestion
    docker tag "${DOCKER_IMG_TAG}" "${DOCKER_IMG_TAG_BRANCH}"
```